harness/tests/common/api_server.py

```diff
@@ -0,0 +1,125 @@
+import contextlib
+import json
+import ssl
+import threading
+from http.server import HTTPServer, SimpleHTTPRequestHandler
+from pathlib import Path
+from typing import Any, Callable, Dict, Iterator, Optional, Tuple
+
+import determined
+from determined.common.api.authentication import salt_and_hash
+
+CERTS_DIR = Path(__file__).parent / "multimaster-certs"
+CERTS1 = {
+    "keyfile": CERTS_DIR / "key1.pem",
+    "certfile": CERTS_DIR / "cert1.pem",
+}
+CERTS2 = {
+    "keyfile": CERTS_DIR / "key2.pem",
+    "certfile": CERTS_DIR / "cert2.pem",
+}
+FIXTURES_DIR = Path(__file__).parent.parent / "fixtures"
+
+
+@contextlib.contextmanager
+def run_api_server(
+    address: Tuple[str, int] = ("localhost", 12345),
+    credentials: Tuple[str, str, str] = ("user1", "password1", "token1"),
+    ssl_keys: Dict[str, Path] = CERTS1,
+) -> Iterator[str]:
+    user, password, token = credentials
+    lock = threading.RLock()
+    state = {}
```

Comment:
there might be a better way of doing this. I'll need to look more into how the HTTP server works with this class. lmk if you have any suggestions